packages/astro/src/assets/build/generate.ts

```diff
@@ -16,18 +16,18 @@ import { getConfiguredImageService } from '../internal.js';
 import type { LocalImageService } from '../services/service.js';
 import type { AssetsGlobalStaticImagesList, ImageMetadata, ImageTransform } from '../types.js';
 import { isESMImportedImage } from '../utils/imageKind.js';
-import { type RemoteCacheEntry, loadRemoteImage } from './remote.js';
+import { type RemoteCacheEntry, loadRemoteImage, revalidateRemoteImage } from './remote.js';
 
 interface GenerationDataUncached {
-	cached: false;
+	cached: CacheStatus.Miss;
 	weight: {
 		before: number;
 		after: number;
 	};
 }
 
 interface GenerationDataCached {
-	cached: true;
+	cached: CacheStatus.Revalidated | CacheStatus.Hit;
```

Comment:
I'm using the third state to include `(revalidated cache entry)` in the output log, since it is a different operation than the two existing statuses. Technically the value isn't read since I'm checking if it isn't "hit" or "miss".

That is really the only thing I'm using the third state for, so if it is not important I can remove it and revert to using a boolean. I think reporting cache revalidation is useful as a third status as it takes longer than a cache hit but uses less bandwidth than a full cache miss.

Reply:
Yeah that's definitely what I understood :) at the end it's just a nitpick, I'm happy with the code as it is